### PR TITLE
Emit error during evaluation or execution phase.

### PR DESCRIPTION
**Pull request by Laimiux**
## What
Let's catch errors and properly emit them as part of Observable callbacks.

### CODE REVIEW

**Comment by carrotkite** [comment]
### JaCoCO Code Coverage 76.19% :white_check_mark:
| Class | Covered | Meta | Status |
|:---|:---:|:---:|:---:|
| `com/instacart/formula/FormulaRuntime` | 84% | 0% | :white_check_mark: |


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by Jawnnypoo — APPROVED** [review]
Would this be considered a breaking change?

**Comment by Laimiux** [comment]
> Would this be considered a breaking change?

Good question, I would say it's not exactly a breaking change - it's more of an improvement. For example, previously by calling `renderModel.onToolbarSelected()` you could unknowingly force a crash in the UI layer. We are now not crashing the UI thread, but passing it to the formula lifecycle owner through the observable callback.

File: formula/src/main/java/com/instacart/formula/FormulaRuntime.kt [hunk 1]
```diff
@@ -116,21 +117,27 @@ class FormulaRuntime<Input : Any, Output : Any>(
...
 
-        if (hasInitialFinished && emitOutput) {
-            emitOutput = false
-            onOutput(checkNotNull(lastOutput))
+            if (hasInitialFinished && emitOutput) {
+                emitOutput = false
+                onOutput(checkNotNull(lastOutput))
+            }
+        } catch (e: Throwable) {
+            manager?.markAsTerminated()
```
Comment (Laimiux) [inline formula/src/main/java/com/instacart/formula/FormulaRuntime.kt:138]: I've added termination of formula when there is an error.

**Pull request merged**